### PR TITLE
Added kolmogorov-Smirnov statistic method to H2OBinomialModelMetrics

### PR DESCRIPTION
**Pull request by Shashank1202**

### CODE REVIEW

**Review by maurever — CHANGES_REQUESTED** [review]
Hi @Shashank1202. Thanks for your contribution! Please add a test case to test the new functionality you added.

**Comment by Shashank1202** [comment]
> Hi @Shashank1202. Thanks for your contribution! Please add a test case to test the new functionality you added.

Hey @maurever would you guide me futher, since it's my 1st OS Contribution. I would be happy to hear from you!

**Comment by Shashank1202** [comment]
> Hi @Shashank1202. Thanks for your contribution! Please add a test case to test the new functionality you added.

Hey @maurever . Thanks for your suggestion and I have added testing regarding this.

**Review by maurever — CHANGES_REQUESTED** [review]
Please remove empty files and config files. 
image

File: h2o-py/tests/pyunit_math/pyunit_ks_metric.py [hunk 1]
```diff
@@ -11,14 +11,21 @@ def kolmogorov_smirnov():
     model.train(x=["Origin", "Distance"], y="IsDepDelayed", training_frame=airlines)
     verify_ks(model, airlines)
 
+    #Test with specific Thresholds
     model = H2OGradientBoostingEstimator(ntrees=1, gainslift_bins=5)
     model.train(x=["Origin", "Distance"], y="IsDepDelayed", training_frame=airlines)
-    ks = model.kolmogorov_smirnov()
+    ks = model.kolmogorov_smirnov(thresholds=[0.01, 0.5, 0.99])
```
[the remark this answers is not in the document]
Comment (Shashank1202) [reply h2o-py/tests/pyunit_math/pyunit_ks_metric.py]: Hey @maurever  
This was my first open-source contribution, and after your review, I have detailedly checked and made the necessary changes accordingly. Could you please go through it once more to ensure everything is in order?

[the remark this answers is not in the document]
Comment (maurever) [reply h2o-py/tests/pyunit_math/pyunit_ks_metric.py]: Hi @Shashank1202. I am happy you are trying to contribute to our open-source library. :+1:  

But your code is not working. I went through your code again and found out the goal is not to add thresholds as a parameter but to implement the kolmogorow_smirnov method on the performance object. So, we can call:

`model.model_performance(data).kolmogorov_smirnov()`

No thresholds are needed. The KS metric is calculated with different thresholds (same for gains lift) than other metrics such as AUC, etc.

File: h2o-py/h2o/model/metrics/binomial.py [hunk 2]
```diff
@@ -976,3 +976,26 @@ def thresholds_and_metric_scores(self):
         if 'thresholds_and_metric_scores' in self._metric_json:
             return self._metric_json['thresholds_and_metric_scores']
         return None
+    
+    def kolmogorov_smirnov(self, thresholds= None):
```
Comment (maurever) [inline h2o-py/h2o/model/metrics/binomial.py:980]:
```suggestion
    def kolmogorov_smirnov(self):
```

File: h2o-py/h2o/model/metrics/binomial.py [hunk 3]
```diff
@@ -976,3 +976,26 @@ def thresholds_and_metric_scores(self):
...
+        >>> response = "economy_20mpg"
+        >>> train, valid = cars.split_frame(ratios = [.8], seed = 1234)
+        >>> cars_gbm = H2OGradientBoostingEstimator(seed = 1234)
+        >>> cars_gbm.train(x = predictors,
+        ...                y = response,
+        ...                training_frame = train,
+        ...                validation_frame = valid)
+        >>> cars_gbm.kolmogorov_smirnov()
+        """
+        return self.metric("ks", thresholds=thresholds)
```
Comment (maurever) [inline h2o-py/h2o/model/metrics/binomial.py:1001]: The goal is something like this:

```suggestion
        return max(self.gains_lift()["kolmogorov_smirnov"])
```

File: h2o-py/tests/pyunit_math/pyunit_ks_metric.py [hunk 4]
```diff
@@ -19,6 +20,21 @@ def kolmogorov_smirnov():
     print(ks_verification)
     assert round(ks, 5) != round(ks_verification, 5)
 
+    # Test with specific thresholds
+    model = H2OGradientBoostingEstimator(ntrees=1, gainslift_bins=5)
+    model.train(x=["Origin", "Distance"], y="IsDepDelayed", training_frame=airlines)
+    ks = model.kolmogorov_smirnov(thresholds=[0.01, 0.5, 0.99])
```
Comment (maurever) [inline h2o-py/tests/pyunit_math/pyunit_ks_metric.py:26]: Have you tried to run the test? This is not working even if you add the method to binomial.py. The goal is not change and test `model.kolmogorov_smirnov()` but add and test `model.model_performance(data).kolmogorov_smirnov()`

```suggestion
```

**Review by maurever — CHANGES_REQUESTED** [review]

**Comment by maurever** [comment]
Hi @Shashank1202 . Thank you for your contribution. We currently don't have the capacity to assist you in completing this task. We decided that this change is not critical for improving our codebase, so we have closed this PR.
